add nametag/hit brightness fix

kurrycat2004 opened the pull request:
- fixes the nametag of spiders, endermen and ender dragons being rendered too dark, because the lightmap coords were getting overwritten and not reset
- fixes spiders, endermen and ender dragons being rendered too red when hit, because the blend function was overwritten and not reset

wlhlm commented:
Thank you for submitting the PR. I was table to manually test this and can confirm this fixes what it says on the tin. However, I have very little knowledge of rendering so I have to defer on others to judge the implementation.

Before:
![2024-11-07_12 05 38](https://github.com/user-attachments/assets/ea9b00e6-aeb2-4e57-96ac-81f832acbd34)

After:
![2024-11-07_13 09 02](https://github.com/user-attachments/assets/cad5fd37-cb19-4519-8e20-4e6a17dd1b6d)

wlhlm commented:
What I'm wondering though, none of the mixins you are adding are specific to the mobs you've mentioned above. Why don't other entities have that problem? Does it really make sense to apply this fix to all entities being rendered, or could we apply this just to the entities mentioned?

kurrycat2004 commented:
this problem only occurs because vanilla changes the gl state in `shouldRenderPass` but never resets it. only endermen, spiders and ender dragons have the specific problem i fixed, because they are the only mobs that render an extra translucent layer for the entity eyes, and therefore setup the blend func and lightmap tex coords for that. both fixes also reset the state to the state it would normally be in for all other mobs, so it wont break anything. i implemented it for every mob in case some mod looks at vanilla code and implements entity eyes the same way.

wlhlm commented:
Thanks for the explanation. That makes sense. I'll defer to @mitchej123 for final approval.

mitchej123 commented:
Thanks for the tag, forgot about this.
